File: finalRest/src/main/java/ajc/formation/project/finalRest/entity/Account.java
package ajc.formation.alten.finalRest.entity;

import java.util.Arrays;
import java.util.Collection;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import com.fasterxml.jackson.annotation.JsonView;

import ajc.formation.alten.finalRest.jsonview.Views;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Pattern;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "ACCOUNT")
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Account implements UserDetails {

    @JsonView(Views.Common.class)
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long idAccount;

    @JsonView(Views.Common.class)
    @NotEmpty
    @Column(name = "LOGIN", nullable = false, unique = true)
    private String login;

    @NotEmpty
    @Pattern(regexp = "^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{4,20}$")
    @Transient
    private String rawPassword;

    @Column(name = "PASSWORD", nullable = false, length = 255)
    private String password;
    
    @JsonView(Views.Common.class)
    @Enumerated(EnumType.STRING)
    @Column(name = "ROLE")
    private AccountRole accountRole;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return Arrays.asList(new SimpleGrantedAuthority(accountRole.toString()));
    }
    @Override
    public String getUsername() {
        return login;
    }
    @Override
    public boolean isAccountNonExpired() {
        return true;
    }
    @Override
    public boolean isAccountNonLocked() {
        return true;
    }
    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }
    @Override
    public boolean isEnabled() {
        return true;
    } 
}

File: finalAng/src/app/model/account.ts
export class Account {
  public get accountRole(): string | undefined {
    return this._accountRole;
  }
  public set accountRole(value: string | undefined) {
    this._accountRole = value;
  }
  public get password(): string | undefined {
    return this._password;
  }
  public set password(value: string | undefined) {
    this._password = value;
  }
  public get login(): string | undefined {
    return this._login;
  }
  public set login(value: string | undefined) {
    this._login = value;
  }
  public get idAccount(): number | undefined {
    return this._idAccount;
  }
  public set idAccount(value: number | undefined) {
    this._idAccount = value;
  }
  constructor(
    private _idAccount?: number,
    private _login?: string,
    private _password?: string,
    private _accountRole?: string
  ) {}
}
